evap/staff/views.py

```diff
@@ -275,6 +275,24 @@ def semester_lottery(request, semester_id):
     template_data =dict(semester=semester, form=form, eligible=eligible, winners=winners)
     return render(request, "staff_semester_lottery.html", template_data)
 
+@staff_required
+def semester_todo(request, semester_id):
+    semester = get_object_or_404(Semester, id=semester_id)
+
+    responsibles = set()
+    prepared_courses = semester.course_set.filter(state__in=['prepared']).all()
+    courses = semester.course_set.filter(state__in=['prepared', 'lecturerApproved']).all()
+
+    for course in prepared_courses:
+        responsibles.add(course.responsible_contributor)
+
+    responsibles = list(responsibles)
```

Comment:
`responsibles = list(set(course.responsible_contributor for course in prepared_courses))`?

or
```
responsibles = (course.responsible_contributor for course in prepared_courses)
responsibles = list(set(responsibles))
```
to be more explicit